src/test/java/fr/paris/lutece/portal/service/daemon/DaemonSchedulerTest.java

```diff
@@ -511,39 +516,39 @@ private void testEnqueueWhileRunning( boolean shouldThrow ) throws ClassNotFound
             assertTrue( scheduler.enqueue( executing, 0L, TimeUnit.MILLISECONDS ) );
             executingDaemon.go( );
             assertTrue( scheduler.enqueue( executing, 0L, TimeUnit.MILLISECONDS ) );
-            try
-            {
-                executingDaemon.go( 200, TimeUnit.MILLISECONDS );
-                fail( "Daemon started execution while already running" );
-            }
-            catch( TimeoutException e )
-            {
-                executingDaemon.resetGo( );
-            }
+
+            executingDaemon.go( 200, TimeUnit.MILLISECONDS );
+            fail( "Daemon started execution while already running" );
+
             executingDaemon.waitForCompletion( );
             executingDaemon.go( );
             executingDaemon.waitForCompletion( );
         }
+        catch(TimeoutException e1)
+        {
+            //ok;
+        }
+        catch( InterruptedException | BrokenBarrierException e)
+        {
+            fail("failed" + e + ":" + e.getMessage( ) );
```

Comment:
here you swallow the exception and you loose the stack trace which might be helpful. Why not let tests throw exceptions, which makes them fail ?